hoomd/hpmc/ExternalFieldLattice.h

```diff
@@ -22,256 +22,278 @@ namespace hoomd
-            }
-        reset(0); // initializes all of the energy parameters.
+        m_k_rotational = k_rotational;
         }
 
-    ~ExternalFieldLattice()
+    //! Getter for rotational spring constant
+    Scalar getKRotational()
         {
-        m_pdata->getBoxChangeSignal()
-            .template disconnect<ExternalFieldLattice<Shape>,
-                                 &ExternalFieldLattice<Shape>::scaleReferencePoints>(this);
+        return m_k_rotational;
         }
 
-    Scalar calculateBoltzmannWeight(uint64_t timestep)
+    //! Helper function to be called when particles are added/removed
+    void slotGlobalParticleNumChange()
         {
-        return 0.0;
+        if (m_lattice_positions.size() != this->m_pdata->getNGlobal()
+                || m_lattice_orientations.size() != this->m_pdata->getNGlobal())
+            {
+            throw std::runtime_error(
+                    "Number of particles no longer equals number of lattice points in ExternalFieldLattice.");
+            }
         }
 
-    double calculateDeltaE(const Scalar4* const position_old_arg,
+    /** Calculate the change in energy for trial moves
+     *
+     * This function currently ignores any information associated with box changes.
+     * However, this function only gets called in the box updater, so it should not ignore box changes.
+     * But why would you have a lattice field and a box updater active in the same simulation?
+     */
+    double calculateDeltaE(const Scalar4* const position_old_arg,  // why is this a Scalar4?
                            const Scalar4* const orientation_old_arg,
                            const BoxDim* const box_old_arg)
         {
-        // TODO: rethink the formatting a bit.
         ArrayHandle<Scalar4> h_pos(m_pdata->getPositions(),
                                    access_location::host,
                                    access_mode::readwrite);
         ArrayHandle<Scalar4> h_orient(m_pdata->getOrientationArray(),
                                       access_location::host,
                                       access_mode::readwrite);
-        const Scalar4* const position_new = h_pos.data;
-        const Scalar4* const orientation_new = h_orient.data;
-        const BoxDim* const box_new = &m_pdata->getGlobalBox();
+        const Scalar4* const position_new = h_pos.data;  // current positions from system definition
+        const Scalar4* const orientation_new = h_orient.data;  // current orientations from system definition
+        //const BoxDim* const box_new = &m_pdata->getGlobalBox();
         const Scalar4 *position_old = position_old_arg, *orientation_old = orientation_old_arg;
-        const BoxDim* box_old = box_old_arg;
+        //const BoxDim* box_old = box_old_arg;
         if (!position_old)
             position_old = position_new;
         if (!orientation_old)
             orientation_old = orientation_new;
-        if (!box_old)
+        /*if (!box_old)
```

Comment:
I left the box stuff commented out here. We either need to remove this stuff or allow this field with variable box sizes (which will take a little more thought about the right way to do).

Reply:
I say remove. Future developers can add scaling back in the correct way if they find a strong use case for it.